### ISSUE TITLE
How to get `wp eval-file` to report errors?

### ISSUE THREAD

**Issue by jplew** [wp-cli/wp-cli#706]
Hi:
I'm a big fan of the `wp eval-file` command. Is there a way to get it report errors to my Terminal? Presently if a script fails it just exits without a peep. For example I want to be alerted for syntax errors or if a variable is undefined.

I have both of these set in my `wp-config.php`:

```
ini_set( 'display_errors', 1 );
define( 'WP_DEBUG_DISPLAY', true );
```

And I've tried both of these to no avail:

```
wp --debug eval-file import-script.php
wp eval-file import-script.php 2>&1
```

Thanks.

**Comment by keithdechant** [comment]
In case anyone is still looking for how to do this, I was able to display errors by adding the following lines to the PHP file being run by wp eval-file:

```
ini_set( 'display_errors', 1 );
error_reporting(E_ALL);
```

**Comment by iandunn** [comment]
@scribu, can you explain the reasoning behind overriding `display_errors` in `\Utils\wp_debug_mode()`? I would expect WP_CLI to respect the `display_errors` value I set via `WP_DEBUG` in `wp-config.php`.

When writing scripts that will be ran with `wp eval-file`, it's inconvenient to have to check an error log whenever the script appears to have failed, rather than having the information immediately available. In many cases, it might not even be obvious that an error occurred at all. It'd also be annoying to have to manually call Core's `wp_debug_mode()` just to re-set `display_errors`, after it's already been called by WP-CLI.

Background information:
* https://github.com/wp-cli/wp-cli/blob/7845a67b97c162573212b6bcf6aff62bfad2957c/php/utils-wp.php#L26
* https://github.com/wp-cli/wp-cli/commit/cb57cc0a7c7d57d595eeb275940d612da12f6522
* https://github.com/wp-cli/wp-cli/pull/786

**Comment by iandunn** [comment]
@danielbachhuber, do you know the answer to https://github.com/wp-cli/wp-cli/issues/706#issuecomment-203610437 ?

**Comment by danielbachhuber** [comment]
> do you know the answer to #706 (comment) ?

Not immediately off the top of my head, no.

**Comment by iandunn** [comment]
Ok, since this is already closed, I created #3220 to discuss whether or not that behavior is desired.
